catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/geofeature/GazetteerFeatureService.java

```diff
@@ -0,0 +1,95 @@
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.catalog.ui.query.geofeature;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.codice.ddf.spatial.geocoder.GeoResult;
+import org.codice.ddf.spatial.geocoder.GeoResultCreator;
+import org.codice.ddf.spatial.geocoding.GeoEntry;
+import org.codice.ddf.spatial.geocoding.GeoEntryQueryException;
+import org.codice.ddf.spatial.geocoding.GeoEntryQueryable;
+import org.opengis.geometry.primitive.Point;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implements a FeatureService using gazetteer(s).
+ * Currently the only gazetteer is provided by the GeoEntryQueryable interface,
+ * which queries geonames.org data. It is intended that more will be added and the results federated.
+ */
+public class GazetteerFeatureService implements FeatureService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GazetteerFeatureService.class);
+
+    private GeoEntryQueryable geoEntryQueryable;
+
+    public void setGeoEntryQueryable(GeoEntryQueryable geoEntryQueryable) {
+        this.geoEntryQueryable = geoEntryQueryable;
+    }
+
+    @Override
+    public List<String> getSuggestedFeatureNames(String query, int maxResults) {
+        List<String> results = new ArrayList<>();
+        try {
+            this.geoEntryQueryable.query(query, maxResults)
+                    .forEach(entry -> results.add(entry.getName()));
+        } catch (GeoEntryQueryException e) {
+            LOGGER.debug("Error while making feature service request.", e);
+        }
+
+        return results;
+    }
+
+    @Override
+    public Feature getFeatureByName(String name) {
+        try {
+            List<GeoEntry> entries = this.geoEntryQueryable.query(name, 1);
+            if (entries.size() > 0) {
+                GeoResult geoResult = getGeoResultFromGeoEntry(entries.get(0));
+                return getFeatureFromGeoResult(geoResult);
+            }
+        } catch (GeoEntryQueryException e) {
+            LOGGER.debug("Error while making feature service request.", e);
+        }
+        return null;
+    }
+
+    protected GeoResult getGeoResultFromGeoEntry(GeoEntry entry) {
+        return GeoResultCreator.createGeoResult(entry);
+    }
+
+    private Feature getFeatureFromGeoResult(GeoResult geoResult) {
+
+        BoundingBoxFeature boundingBoxFeature = new BoundingBoxFeature();
+        boundingBoxFeature.setName(geoResult.getFullName());
+
+        List<Point> bbox = geoResult.getBbox();
```

Comment:
Should we check is the point list is null or empty here?  `CollectionUtils` has some handy methods for that if so